cfme/automate/dialogs/service_dialogs.py

```diff
@@ -0,0 +1,161 @@
+from widgetastic_patternfly import CandidateNotFound
+from cached_property import cached_property
+
+from cfme.exceptions import ItemNotFound
+from cfme.modeling.base import BaseCollection, BaseEntity
+from cfme.utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
+
+from . import AutomateCustomizationView, AddDialogView, EditDialogView
+from .dialog_tab import TabCollection
+
+
+class DialogsView(AutomateCustomizationView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and
+            self.title.text == 'All Dialogs' and
+            self.service_dialogs.is_opened and
+            self.service_dialogs.tree.currently_selected == ["All Dialogs"])
+
+
+class DetailsDialogView(AutomateCustomizationView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and self.service_dialogs.is_opened and
+            self.title.text == 'Dialog "{}"'.format(self.context['object'].label)
+        )
+
+
+@attr.s
+class Dialog(BaseEntity, Fillable):
+    """A class representing one Dialog in the UI."""
+    label = attr.ib()
+    description = attr.ib(default=None)
+
+    _collections = {'tabs': TabCollection}
+
+    def as_fill_value(self):
+        return self.label
+
+    @property
+    def dialog(self):
+        return self
+
+    @cached_property
+    def tabs(self):
+        return self.collections.tabs
+
+    @property
+    def tree_path(self):
+        return self.parent.tree_path + [self.label]
+
+    def update(self, updates):
+        """ Update dialog method"""
+        view = navigate_to(self, 'Edit')
+        changed = view.fill(updates)
+        if changed:
+            view.save_button.click()
+        else:
+            view.cancel_button.click()
+        view = self.create_view(DetailsDialogView, override=updates)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+        if changed:
+            view.flash.assert_message(
+                '{} was saved'.format(updates.get('name', self.label)))
+        else:
+            view.flash.assert_message(
+                'Dialog editing was canceled by the user.')
+
+    def delete(self):
+        """ Delete dialog method"""
+        view = navigate_to(self, "Details")
+        view.configuration.item_select('Remove Dialog', handle_alert=True)
+        view = self.create_view(DialogsView)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+        view.flash.assert_success_message(
+            'Dialog "{}": Delete successful'.format(self.label))
+
+    @property
+    def exists(self):
+        """ Returns True if dialog exists"""
+        try:
+            navigate_to(self, 'Details')
+            return True
+        except (CandidateNotFound, ItemNotFound):
+            return False
+
+    def delete_if_exists(self):
```

Comment:
I'd call this a convenience method, and would request it's removed in favor of an optional argument to `delete` method to check for exists first, or removed altogether and left to test fixtures.

Reply:
@psav , @mfalesni : what do you think ? 
Some places delete are used for multiple delete. 
So we can use arg delete(if_exist=False) .